cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -516,6 +516,22 @@ var (
 					device.DevEui = devID.DevEui
 				}
 			}
+			is, err := api.Dial(ctx, config.IdentityServerGRPCAddress)
+			if err != nil {
+				return err
+			}
+			requestDevEUI, _ := cmd.Flags().GetBool("request-dev-eui")
+			if requestDevEUI {
+				logger.Debug("request-dev-eui flag set, requesting a DevEUI from the block for the application")
+				appID := getApplicationID(cmd.Flags(), args)
```

Comment:
`nit`: I think this is already in `devID`, and maybe the EUI should be filled there too.